### ISSUE TITLE
Remove operator from reactor

### ISSUE THREAD

**Issue by jakehader** [terrapower/armi#830]
There are only a handful of uses of `r.o`. it would be great to get rid of these, as:

- it is unclear why a reactor should have an operator
- reactor construction is muddled by being incomplete until the reactor has an operator
- many instances are only used for getting a `cs`
- removing the `r.o` would reduce the number of circular references `r.o` -> `o.r` -> `r.o` ...
- in almost all circumstances, a reactor and an operator are already in scope.

**Comment by john-science** [comment]
Yeah, it's never made sense to me that the `Reactor` object should have an `Operator` on it.

It really seems like accidental design, not purposeful design.

**Comment by john-science** [comment]
Here are all of the usages of this I could find in ARMI (though, of course, any number of plugins might be using this "feature"):

```python
armi/interfaces.py:  self.o = r.o if r else None
armi/mpiActions.py:  self.r.o = self.o

armi/operators/operator.py:  r.o = self
armi/operators/operator.py:  self.r.o = self
armi/operators/operator.py:  self.r.o = None

armi/physics/neutronics/globalFlux/globalFluxInterface.py:  daysIntoCycle = sum(self.r.o.stepLengths[cycle][:timeNode])
armi/physics/neutronics/macroXSGenerationInterface.py:  sum([self.r.o.burnSteps[i] + 1 for i in range(cycle)]) + self.r.p.timeNode

armi/reactor/converters/geometryConverters.py:  self._o = self._sourceReactor.o
armi/reactor/converters/parameterSweeps/generalParameterSweepConverters.py:  fh = r.o.getInterface("fuelHandler")

armi/reactor/converters/uniformMesh.py:  newReactor = Reactor(sourceReactor.o.cs.caseTitle, bp)
armi/reactor/converters/uniformMesh.py:  coreDesign.construct(sourceReactor.o.cs, bp, newReactor, loadAssems=False)
armi/reactor/converters/uniformMesh.py:  _geometryConverter = newReactor.core.growToFullCore(sourceReactor.o.cs)

armi/utils/reportPlotting.py:  dbi = reactor.o.getInterface("database")
armi/utils/reportPlotting.py:  dd = reactor.o.getInterface("ductDistortion")
armi/utils/reportPlotting.py:  history = reactor.o.getInterface("history")
```

**Comment by john-science** [comment]
@jakehader  So far, there are no downstream impacts. I only did the removed the really low-hanging fruit versions of `r.o` (that I could fit into my day).

What's left is some higher-level stuff that will have possible downstream impacts. It appears that sometimes people use `r.o.getInterface("xxx")`, and squashing all usages of those is the trick.  There aren't TOO many places people do this, but it's certainly some fast-and-loose code that will take some untangling to improve the code flow.

TLDR; this is just a start. Thanks!

**Comment by john-science** [comment]
Bump. I still consider this ticket important.

**Comment by yardasol** [comment]
@yardasol is interested in this one
